### PR TITLE
Issue 4 fixed + some extra features

### PR DESCRIPTION
**Pull request by eatcha-wikimedia**
Changes:

**1** https://github.com/Zitrax/FPCBot/issues/4 fixed

**2** Tries to get the Category from nomination page

**3** Current month page changed , users had to manually clear [this](https://commons.wikimedia.org/wiki/Commons:Featured_pictures/chronological/current_month) page at the end of month.

**4** Add small s/o/n counts in the montly featured page.

### CODE REVIEW

**Comment by eatcha-wikimedia** [comment]
Maybe https://github.com/Zitrax/FPCBot/issues/4 should be closed now.

File: fpc.py [hunk 1]
```diff
@@ -136,9 +136,33 @@ def uploader(self, link=True):
             return history[0][2]
 
     def creator(self):
-        """Return the link to the user that created the image"""
+        """Return the link to the user that created the image."""
         return self.uploader()
 
+    def FindCategoryOfFile(self):
```
Comment (Zitrax) [inline fpc.py]: Minor, but existing functions start with lowercase letters.

File: fpc.py [hunk 2]
```diff
@@ -136,9 +136,33 @@ def uploader(self, link=True):
...
+
+    def NewFileNameIfMoved(self):
+        """Fixed https://github.com/Zitrax/FPCBot/issues/4."""
+        page = pywikibot.Page(G_Site, self.fileName())
+        if page.isRedirectPage() == True:
+            newfilename= re.sub (r'(?:\[|\]|commons:)', '', str(page.getRedirectTarget()))
+            return newfilename
+        else:
+            file_name = self.fileName()
+            return file_name
```
Comment (Zitrax) [inline fpc.py]: Could return directly without assigning to newfilename.

**Comment by eatcha-wikimedia** [comment]
Thanks for reviewing, I fixed another issue with uploader's name it should prevent the bot from notifying the file renamer with promotion template.

**Comment by eatcha-wikimedia** [comment]
@Zitrax ?

**Comment by Zitrax** [comment]
Looks good to me. Ready to merge?

**Comment by eatcha-wikimedia** [comment]
Wait, the folks at [fpc are renaming fpc category to fpc gallery.](https://commons.wikimedia.org/w/index.php?title=User_talk:Eatcha&oldid=396624380#Some_new_things_on_FP) I should fix the category detection regex before merging and check if any other changes are required. I will notify you when the changes are done. I was notified tomorrow about the change.

**Comment by eatcha-wikimedia** [comment]
@Zitrax I closed and park 2 nominations with the new source and I think can be merged now.

**Comment by eatcha-wikimedia** [comment]
I mean "add images to the section directly" if the section is specified in nomination page.

**Comment by Zitrax** [comment]
Feel free to just merge it when done.

**Comment by eatcha-wikimedia** [comment]
@Zitrax  I don't have the write access anymore. I deactivated my account for 1 week and now I don't have access to this project and all other projects I could write into. 
Is it possible to send me a second  letter of invite ? . I really appreciate it.

Here's the direct link to access settings:
https://github.com/Zitrax/FPCBot/settings/access

**Comment by Zitrax** [comment]
No problem, I sent you an invite now.

**Comment by eatcha-wikimedia** [comment]
Thanks.

**Pull request merged**